packages/lib-panoptes-js/src/panoptes.js

```diff
@@ -52,22 +53,49 @@ function getQueryParams (query) {
   }
 }
 
+/*
+Converts a query object (e.g. { foo: 'bar', img: 'http://foo.bar/baz.jpg' })
+into a query string (e.g. "foo=bar&img=http%3A%2F%2Ffoo.bar%")
+ */
+function getQueryString (query, endpoint = '') {
+  const queryParams = getQueryParams(query)
+  let queryString = Object.entries(queryParams)
+    .filter(([key, val]) => val !== undefined)
+    .map(([key, val]) => `${encodeURIComponent(key)}=${encodeURIComponent(val)}`)
+    .join('&')
+  
+  if (queryString.length > 0) {
+    return endpoint.includes('?')
+      ? `&${queryString}`  // If endpoint already has a query '?', prepend the query string with a '&'.
+      : `?${queryString}`  //  Otherwise, prepend with a '?'.
+  }
+  return ''
+}
+
 // TODO: Consider how to integrate a GraphQL option
 function get (endpoint, query = {}, headers = {}, host) {
   if (!endpoint) return handleMissingParameter('Request needs a defined resource endpoint')
   if (typeof query !== 'object') return Promise.reject(new TypeError('Query must be an object'))
 
   const apiHost = determineHost(query, host)
-  const request = superagent.get(`${apiHost}${endpoint}`)
-    .set('Content-Type', 'application/json')
-    .set('Accept', 'application/vnd.api+json; version=1')
-
-  if (headers){
-    request.set(parseHeaders(headers))
-  }
-  const queryParams = getQueryParams(query)
-
-  return request.query(queryParams).then(response => response)
+  const queryString = getQueryString(query, endpoint)
+  const fetchHeaders = Object.assign({
```

Comment:
`Object.assign` has been replaced by the rest spread operator, which is much faster, in modern JavaScript.

Reply:
`Object.assign` should only appear in code if you need to support IE.